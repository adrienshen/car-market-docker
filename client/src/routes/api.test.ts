// require('dotenv').config();
import * as request from 'supertest';
import server from '../app';

console.log('Environment: ', process.env.CAR_SERVICE_URL);

// TODO: improvement: mock the car-service since it's already tested in the micro-service code
jest.setTimeout(15000)

test('GET: /v1/cars/:id returns correct data', () => {
  return request(server).get('/api/v1/cars/JHk290Xj')
    .then(response => {
      expect(response.status).toBe(200);
      expect(response.text).toMatchSnapshot();
      expect(response.body.car.id).toBe('JHk290Xj');
      expect(response.body.message).toBe('2010 Ford F10 with 120123 miles available for $19,999.00 listed since Sun Jun 20 2021 13:58:46 GMT-0500 (Central Daylight Time)');
    })
});

test('GET: /v1/cars?make=Toyota returns correct data', () => {
  return request(server).get('/api/v1/cars?make=Toyota').then(response => {
    // console.log('response: ', response.body);
    expect(response.status).toBe(200);
    expect(response.body.results.length).toBe(2);
    expect(response.text).toMatchSnapshot();
  })
});

test('GET: /v1/cars?year=2018&mileage_gt=24000 returns correct data', () => {
  return request(server).get('/api/v1/cars?year=2018&mileage_gt=24000').then(response => {
    console.log('3. response: ', response.body);
    expect(response.status).toBe(200);
    expect(response.body.results.length).toBe(1);
    expect(response.text).toMatchSnapshot();
    expect(response.body.results[0].id).toBe('1i3xjRllc');
  })
});
