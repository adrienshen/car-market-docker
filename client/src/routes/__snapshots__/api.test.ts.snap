// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET: /v1/cars/:id returns correct data 1`] = `"{\\"message\\":\\"2010 Ford F10 with 120123 miles available for $19,999.00 listed since Sun Jun 20 2021 13:58:46 GMT-0500 (Central Daylight Time)\\",\\"car\\":{\\"id\\":\\"JHk290Xj\\",\\"make\\":\\"Ford\\",\\"model\\":\\"F10\\",\\"package\\":\\"Base\\",\\"color\\":\\"Silver\\",\\"year\\":2010,\\"category\\":\\"Truck\\",\\"mileage\\":120123,\\"price_cents\\":1999900,\\"created_at\\":\\"2021-06-20T18:58:46.170Z\\",\\"updated_at\\":\\"2021-06-20T18:58:46.170Z\\"}}"`;

exports[`GET: /v1/cars/:id works 1`] = `"{\\"message\\":\\"2010 Ford F10 with 120123 miles available for $19,999.00 listed since Sun Jun 20 2021 13:58:46 GMT-0500 (Central Daylight Time)\\",\\"car\\":{\\"id\\":\\"JHk290Xj\\",\\"make\\":\\"Ford\\",\\"model\\":\\"F10\\",\\"package\\":\\"Base\\",\\"color\\":\\"Silver\\",\\"year\\":2010,\\"category\\":\\"Truck\\",\\"mileage\\":120123,\\"price_cents\\":1999900,\\"created_at\\":\\"2021-06-20T18:58:46.170Z\\",\\"updated_at\\":\\"2021-06-20T18:58:46.170Z\\"}}"`;

exports[`GET: /v1/cars?make=Toyota returns correct data 1`] = `"{\\"message\\":\\"There were 2 founded with your filters\\",\\"results\\":[{\\"id\\":\\"fWI37la\\",\\"make\\":\\"Toyota\\",\\"model\\":\\"Camry\\",\\"package\\":\\"SE\\",\\"color\\":\\"White\\",\\"year\\":2019,\\"category\\":\\"Sedan\\",\\"mileage\\":3999,\\"price_cents\\":2899000,\\"created_at\\":\\"2021-06-20T18:58:46.170Z\\",\\"updated_at\\":\\"2021-06-20T18:58:46.170Z\\"},{\\"id\\":\\"1i3xjRllc\\",\\"make\\":\\"Toyota\\",\\"model\\":\\"Rav4\\",\\"package\\":\\"XSE\\",\\"color\\":\\"Red\\",\\"year\\":2018,\\"category\\":\\"SUV\\",\\"mileage\\":24001,\\"price_cents\\":2275000,\\"created_at\\":\\"2021-06-20T18:58:46.170Z\\",\\"updated_at\\":\\"2021-06-20T18:58:46.170Z\\"}]}"`;

exports[`GET: /v1/cars?make=Toyota returns correct data 2`] = `"{\\"message\\":\\"There were 2 founded with your filters\\",\\"results\\":[{\\"id\\":\\"fWI37la\\",\\"make\\":\\"Toyota\\",\\"model\\":\\"Camry\\",\\"package\\":\\"SE\\",\\"color\\":\\"White\\",\\"year\\":2019,\\"category\\":\\"Sedan\\",\\"mileage\\":3999,\\"price_cents\\":2899000,\\"created_at\\":\\"2021-06-20T18:58:46.170Z\\",\\"updated_at\\":\\"2021-06-20T18:58:46.170Z\\"},{\\"id\\":\\"1i3xjRllc\\",\\"make\\":\\"Toyota\\",\\"model\\":\\"Rav4\\",\\"package\\":\\"XSE\\",\\"color\\":\\"Red\\",\\"year\\":2018,\\"category\\":\\"SUV\\",\\"mileage\\":24001,\\"price_cents\\":2275000,\\"created_at\\":\\"2021-06-20T18:58:46.170Z\\",\\"updated_at\\":\\"2021-06-20T18:58:46.170Z\\"}]}"`;

exports[`GET: /v1/cars?year=2018&mileage_gt=24000 returns correct data 1`] = `"{\\"message\\":\\"There were 1 founded with your filters\\",\\"results\\":[{\\"id\\":\\"1i3xjRllc\\",\\"make\\":\\"Toyota\\",\\"model\\":\\"Rav4\\",\\"package\\":\\"XSE\\",\\"color\\":\\"Red\\",\\"year\\":2018,\\"category\\":\\"SUV\\",\\"mileage\\":24001,\\"price_cents\\":2275000,\\"created_at\\":\\"2021-06-20T18:58:46.170Z\\",\\"updated_at\\":\\"2021-06-20T18:58:46.170Z\\"}]}"`;
